src/main/java/com/powsybl/openloadflow/network/LfContingency.java

```diff
@@ -75,6 +92,10 @@ public int getIndex() {
         return index;
     }
 
+    public int getNbSynchronousComponents() {
```

Comment:
I think that here you have to store directly the number of components created by the contingency and add the -1 in that class.